src/operator/operator_common.h

```diff
@@ -253,6 +284,48 @@ inline bool type_assign(int *y, const int& x) {
   }
 #endif
 
+/*! \brief assign stype to target_stype, if successful,
+ *         assign dispatch_mode to target_dispatch
+ */
+inline bool dispatch_on_storage(int* stype,
```

Comment:
why isn't this called storage_type_assign?

Reply:
It also tries to assign dispatch_mode when stype assign is successful. The behaviour is a bit different from type_assign and shape_assign. I named it differently to avoid confusion. Does taht make sense?

Reply:
I think it's better to just name it storage_type_assign